website/addons/googledrive/tests/test_models.py

```diff
@@ -131,296 +131,362 @@ def test_node_addon_get_or_create_finds_changed_root(self):
+        assert_equal(res['profile_url'], 'fakeUrl')
 
-    def test_has_auth(self):
-        user_settings = GoogleDriveUserSettingsFactory(access_token=None)
-        assert_false(user_settings.has_auth)
-        user_settings.access_token = '12345'
-        user_settings.save()
-        assert_true(user_settings.has_auth)
 
-    @mock.patch.object(GoogleDriveOAuthSettings, '_needs_refresh')
-    def test_access_token_checks(self, mock_needs_refresh):
-        mock_needs_refresh.return_value = False
-        user_settings = GoogleDriveUserSettingsFactory()
-
-        user_settings.access_token
+    @mock.patch.object(GoogleAuthClient, 'refresh')
+    def test_refresh_token(self, mock_refresh):
+        fake_access_token = 'abc123'
+        fake_refresh_token = 'xyz456'
+        mock_refresh.return_value = 'faketoken'
 
-        mock_needs_refresh.assert_called_once()
+        res = self.provider._refresh_token(fake_access_token, fake_refresh_token)
+        assert_equal(res, 'faketoken')
 
-    @mock.patch.object(GoogleAuthClient, 'refresh')
-    @mock.patch.object(GoogleDriveOAuthSettings, '_needs_refresh')
-    def test_access_token_refreshes(self, mock_needs_refresh, mock_refresh):
-        mock_refresh.return_value = {
-            'access_token': 'abc',
-            'refresh_token': '123',
-            'expires_at': time.time(),
-        }
-        user_settings = GoogleDriveUserSettingsFactory()
-        user_settings.expires_at = datetime.now()
-        user_settings.access_token
-        mock_refresh.assert_called_once()
 
     @mock.patch.object(GoogleAuthClient, 'refresh')
-    def test_access_token_refreshes_timeout(self, mock_refresh):
-        mock_refresh.return_value = {
-            'access_token': 'abc',
-            'refresh_token': '123',
-            'expires_at': time.time(),
-        }
-        user_settings = GoogleDriveUserSettingsFactory()
-        user_settings.expires_at = (datetime.utcnow() + relativedelta.relativedelta(seconds=5))
+    def test_refresh_token_without_refresh_token(self, mock_refresh):
+        fake_access_token = 'abc123'
+        fake_refresh_token = None
+        mock_refresh.return_value = 'faketoken'
 
-        user_settings.access_token
+        res = self.provider._refresh_token(fake_access_token, fake_refresh_token)
+        assert_false(res, 'faketoken')
 
-        mock_refresh.assert_called_once()
+class TestGoogleDriveUserSettings(OsfTestCase):
 
-    @mock.patch.object(GoogleAuthClient, 'refresh')
-    def test_access_token_refreshes_timeout_longer(self, mock_refresh):
-        mock_refresh.return_value = {
-            'access_token': 'abc',
-            'refresh_token': '123',
-            'expires_at': time.time(),
-        }
-        user_settings = GoogleDriveUserSettingsFactory()
-        user_settings.expires_at = datetime.utcnow() + relativedelta.relativedelta(minutes=4)
-        user_settings.access_token
-        mock_refresh.assert_called_once()
+    def _prep_oauth_case(self):
```

Comment:
Every single test in this class appears to run this function. Perhaps this should just be the `setUp` method instead?